src/test/java/endtoend/ExceptionHandlingEndToEndTest.java

```diff
@@ -12,15 +14,15 @@
     public ExpectedException expected = ExpectedException.none();
     
     @Test
-    public void error404IsCorrectlyHandled() {
-        expected.expect(NotFoundException.class);
-        this.performGETRequest(Object.class, "bogus");
+    public void error401IsCorrectlyHandled() {
```

Comment:
The tests are now ordered by error status code, to make finding the correct test case easier.